### PR TITLE
Add Experian check rule action

### PR DESCRIPTION
**Pull request by VladimirBerdnik**

### CODE REVIEW

File: openapi/components/schemas/Rules/RuleAction.yaml [hunk 1]
```diff
@@ -28,27 +29,28 @@ properties:
...
+      - add-risk-score
+      - adjust-ready-to-pay
       - blocklist
       - cancel-scheduled-payments
+      - create-infusionsoft-order
+      - create-infusionsoft-payment
+      - display-other-choices
       - guess-payment-card-expiration
+      - offer-purchase-bump
+      - perform-experian-check
```
Comment (VladimirBerdnik) [inline openapi/components/schemas/Rules/RuleAction.yaml:41]: added this line and sorted the others alphabetically

**Review by ashkarpetin — DISMISSED** [review]

File: openapi/components/schemas/Rules/RuleAction.yaml (hunk 1 above)
Comment (slavcodev) [reply openapi/components/schemas/Rules/RuleAction.yaml:41]: Other idea for rule/decorator? To sort enums?

**Review by slavcodev — DISMISSED** [review]

File: openapi/components/schemas/Rules/Actions/perform-experian-check.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+description: Perform Experian ProveID check on the customer
+allOf:
+  - $ref: ../RuleAction.yaml
+  - type: object
+    properties:
+      experianCredentialHash:
+        description: Hash of the Experian credential to use for performing the check.
+        type: string
```
Comment (slavcodev) [inline openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:8]: I wonder should it be string or list of strings?

File: openapi/components/schemas/Rules/Actions/perform-experian-check.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        example: 58816e40-387e-4834-982f-fa1855a75f64
+      tagForApprovedIdentity:
+        description: A tag ID that should be used to tag the customer when the identity matches Experian records.
+        type: string
+      tagForRejectedIdentity:
+        description: A tag ID that should be used to tag the customer when the identity doesn't match Experian records.
+        type: string
```
Comment (slavcodev) [inline openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:15]: I wonder should it be string or list of strings?

File: openapi/components/schemas/Rules/Actions/perform-experian-check.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+description: Perform Experian ProveID check on the customer
+allOf:
+  - $ref: ../RuleAction.yaml
+  - type: object
+    properties:
+      experianCredentialHash:
+        description: Hash of the Experian credential to use for performing the check.
+        type: string
+        example: 58816e40-387e-4834-982f-fa1855a75f64
+      tagForApprovedIdentity:
```
Comment (slavcodev) [inline openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:10]: approvalTag(s), declineTag(s), unknownResultTag(s) ?

File: openapi/components/schemas/Rules/Actions/perform-experian-check.yaml (hunk 3 above)
Comment (VladimirBerdnik) [reply openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:15]: I thought about arrays, but they seems excessive

File: openapi/components/schemas/Rules/Actions/perform-experian-check.yaml (hunk 4 above)
Comment (VladimirBerdnik) [reply openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:10]: `approvalTag` sounds like `A tag that approves`

Comment (slavcodev) [reply openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:10]: is not it approve the customer?

Comment (VladimirBerdnik) [reply openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:10]: But not the tag approves. Experian approves, the check approves. The tag is the result and not the reason, as I see it

Comment (slavcodev) [reply openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:10]: tag approves the customer to be used in our application as approved :)

Comment (slavcodev) [reply openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:10]: it's just suggestion, I'm trying to make them simpler. may be
```
resultTags:
  approved: tag1 
  declined: tag2
  unknown: tag3
```

File: openapi/components/schemas/Rules/Actions/perform-experian-check.yaml (hunk 3 above)
Comment (slavcodev) [reply openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:15]: It seems does not complecate neither backend, not frontend, but open some possibilities.
For example, I imagine that I might want to add `experian` + `experian-approved` tags for approval, to easier filter customers.

Comment (VladimirBerdnik) [reply openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:15]: there is existing action `Tag or untag customer`. We may select **Tag customer with `Experian` tag** and **Perform Experian check, tag as `Experian-approved` if approved**

Comment (slavcodev) [reply openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:15]: Can you rephrase the latest comment? I think I don't understand it.

Comment (VladimirBerdnik) [reply openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:15]: image

Comment (slavcodev) [reply openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:15]: You can't do this

On transaction process requested:

- add tag "experian"
- if "tag !== experian" then perform eperian check

All actions in one rule are considered async and the order is not quaranteed.

Comment (VladimirBerdnik) [reply openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:15]: in your example `experian` tag will also be added after the check

Comment (slavcodev) [reply openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:15]: How?

Comment (VladimirBerdnik) [reply openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:15]: Also, I think the order is following:

if "tag !== experian" then
- add tag "experian"
- perform eperian check

Comment (slavcodev) [reply openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:15]: Ohh you mean two actions in the same rule, I thought it's about different rules. That makes sense now. Thanks.

File: openapi/components/schemas/Rules/Actions/perform-experian-check.yaml (hunk 4 above)
Comment (VladimirBerdnik) [reply openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:10]:
```
tagOnApprove:
tagOnReject:
tagOnUnknown:
```

Comment (slavcodev) [reply openapi/components/schemas/Rules/Actions/perform-experian-check.yaml:10]: LGTM

**Review by slavcodev — APPROVED** [review]

**Comment by VladimirBerdnik** [comment]
php please review

**Comment by VladimirBerdnik** [comment]
php please review

**Review by ashkarpetin — APPROVED** [review]
